### PR TITLE
Page docs improvements

### PR DESCRIPTION
**Pull request by kblok**
* Start generating the XML document so we get warnings.
* Fix broken XML comments
* Add missing comments

related to #236

### CODE REVIEW

**Comment by kblok** [comment]
@Meir017

File: lib/PuppeteerSharp/Page.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
         #region Public Properties
 
         /// <summary>
-        /// Raised when the JavaScript <c>load</c> <see cref="https://developer.mozilla.org/en-US/docs/Web/Events/load"/> event is dispatched.
+        /// Raised when the JavaScript <c>load</c> <see Hef="https://developer.mozilla.org/en-US/docs/Web/Events/load"/> event is dispatched.
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: I think you meant `href` 😉

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -463,7 +469,7 @@ public Task<JSHandle> QuerySelectorAllHandleAsync(string selector)
         }
 
         /// <summary>
-        /// Adds a <c><script></c> tag into the page with the desired url or content
+        /// Adds a <c><![CDATA[<script>]]></c> tag into the page with the desired url or content
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: is this needed?

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -473,7 +479,7 @@ public Task<JSHandle> QuerySelectorAllHandleAsync(string selector)
         public Task<ElementHandle> AddScriptTagAsync(AddTagOptions options) => MainFrame.AddScriptTag(options);
 
         /// <summary>
-        /// Adds a <c><script></c> tag into the page with the desired url or content
+        /// Adds a <c><![CDATA[<script>]]></c> tag into the page with the desired url or content
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: is this needed?

File: lib/PuppeteerSharp/Page.cs [hunk 4]
```diff
@@ -759,9 +780,19 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
             return new MemoryStream(buffer);
         }
 
+        /// <summary>
+        /// Enables/Disables Javascript on the page
+        /// </summary>
+        /// <returns>Task.</returns>
+        /// <param name="enabled">Whether or not to enable JavaScript on the page.</param>
         public async Task SetJavaScriptEnabledAsync(bool enabled)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: remove the `async` & `await`

File: lib/PuppeteerSharp/Page.cs [hunk 5]
```diff
@@ -759,9 +780,19 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
...
             => await Client.SendAsync("Emulation.setScriptExecutionDisabled", new { value = !enabled });
 
+        /// <summary>
+        /// Emulates a media such as screen or print.
+        /// </summary>
+        /// <returns>Task.</returns>
+        /// <param name="media">Media to set.</param>
         public async Task EmulateMediaAsync(MediaType media)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: remove the `async` & `await`

File: lib/PuppeteerSharp/Page.cs [hunk 6]
```diff
@@ -783,6 +814,16 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
             }
         }
 
+        /// <summary>
+        /// Emulates given device metrics and user agent. 
+        /// </summary>
+        /// <remarks>
+        /// This method is a shortcut for calling two methods:
+        /// page.setUserAgent(userAgent)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: this is the js syntax 😉

File: lib/PuppeteerSharp/Page.cs [hunk 7]
```diff
@@ -1591,6 +1636,13 @@ string SerializeArgument(object arg)
         #endregion
 
         #region IDisposable
+        /// <summary>
+        /// Releases all resource used by the <see cref="T:PuppeteerSharp.Page"/> object by calling the <see cref="CloseAsync"/> method.
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: What is the `T:` prefix?

File: lib/PuppeteerSharp/Page.cs (hunk 2 above)
Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Yes. `<script>` breaks the XML so it would be ignored by the compiler.

File: lib/PuppeteerSharp/Page.cs (hunk 3 above)
Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Same

File: lib/PuppeteerSharp/Page.cs (hunk 7 above)
Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Changed, was generated by the IDE.

**Comment by kblok** [comment]
Done @Meir017

**Pull request merged**